tasks/github_tasks.py

```diff
@@ -188,3 +189,60 @@ def get_token_from_app(_, app_id_env='GITHUB_APP_ID', pkey_env='GITHUB_KEY_B64')
     from .libs.common.github_api import GithubAPI
 
     GithubAPI.get_token_from_app(app_id_env, pkey_env)
+
+
+@task
+def assign_team_label(_, pr_id=-1):
+    """
+    Assigns the github team label name if a single team can
+    be deduced from the changed files
+    """
+    from .libs.common.github_api import GithubAPI
+
+    def get_team(changed_files) -> str | None:
+        codeowners = read_owners('.github/CODEOWNERS')
+
+        global_team = None
+        for file in changed_files:
+            owners = [name for (kind, name) in codeowners.of(file) if kind == 'TEAM']
+            if len(owners) != 1:
```

Comment:
Rather than dismissing the multi-owner case I think we'd rather compute the number of occurrence of each owner and assign to the greatest one